packages/babel-plugin-transform-react-constant-elements/test/fixtures/constant-elements/member-expression-this/output.js

```diff
@@ -1,14 +1,14 @@
-var _ref = <span>Sub Component</span>;
+var _span;
 
 class Component extends React.Component {
   constructor(...args) {
-    super(...args);
+    var _this$subComponent;
 
-    this.subComponent = () => _ref;
+    super(...args);
 
-    var _ref2 = <this.subComponent />;
+    this.subComponent = () => _span || (_span = <span>Sub Component</span>);
 
-    this.render = () => _ref2;
+    this.render = () => _this$subComponent || (_this$subComponent = <this.subComponent />);
```

Comment:
I'm not sure I like caching this. For now let's keep the current behavior since no one complained, but this doesn't look safe (I don't know how "react components on `this` are usually used).